### PR TITLE
Docs: Use "fiber blocking" instead of "semantic blocking"

### PR DESCRIPTION
**Pull request by Daenyth**
From the discord discussions, everyone seems to like this phrasing better - it
helps avoid confusion for newcomers, about the difference between thread
blocking and fiber blocking

### CODE REVIEW

File: docs/thread-model.md [hunk 1]
```diff
@@ -104,40 +105,40 @@ unbounded, cached threadpool and then shift computation back to the compute pool
...
 
-### Semantic blocking
+### Fiber blocking
 
 Of course, we do also need the ability to tell fibers to wait for conditions to
 be fulfilled. If we can't call thread blocking operations (eg Java/Scala builtin
 locks, semaphores, etc) then what can we do? It seems we need a notion of
 _semantic_ blocking, where the execution of a fiber is suspended and control of
-the thread it was running on is yielded.
+the thread it was running on is yielded. This concept is called "fiber blocking" in cats-effect.
```
Comment (armanbilge) [inline docs/thread-model.md:114]: We should make the connection to the old name, something like this I reckon.

```suggestion
the thread it was running on is yielded. This concept is called "fiber blocking" in cats-effect (previously described as "semantic blocking").
```

Comment (Daenyth) [reply docs/thread-model.md:114]: Do you think that's still needed in the context of the previous sentence?

> It seems we need a notion of
_semantic_ blocking, where the execution of a fiber is suspended and control of
the thread it was running on is yielded. This concept is called "fiber blocking" in cats-effect.

Comment (armanbilge) [reply docs/thread-model.md:114]: Oh, missed that, sorry. Hm. I think it's still good to call this out explicitly, like as if we renamed a method. Maybe even in the section title.

File: docs/thread-model.md [hunk 2]
```diff
@@ -334,16 +336,16 @@ the event that the fiber is canceled.
...
+* This means that we can implement thread affinity, where a fiber that yields is most
    likely to be re-scheduled on the same thread. This makes yielding much cheaper
    as if the fiber is immediately re-scheduled we don't even have to flush CPU caches
- * Consequently we can support auto-yielding where a fiber will insert an `IO.cede`
+* Consequently we can support auto-yielding where a fiber will insert an `IO.cede`
    every fixed number of iterations of the runloop, stopping a rogue cpu-bound fiber
    from inadvertently pinning a CPU core
 
-## And that's it!
+## And that's it
```
Comment (armanbilge) [inline docs/thread-model.md]: Don't like exclamations? 😆

Comment (Daenyth) [reply docs/thread-model.md]: Accidental

Comment (Daenyth) [reply docs/thread-model.md]: Actually I see now it's vscode's markdown formatter removing it automatically - apparently trailing punctuation is a lint

**Review by armanbilge — APPROVED** [review]

**Review by djspiewak — APPROVED** [review]

**Pull request merged**